documentation/metering/metering.md

```diff
@@ -0,0 +1,154 @@
+
+| Attribute | Type | Default Value | Description |
+| --- | --- | --- | --- |
+| SAI_METER_BUCKET_ATTR_ENI_ID | `sai_object_id_t` | SAI_NULL_OBJECT_ID | ENI ID of this metering class. |
+| SAI_METER_BUCKET_ATTR_METER_CLASS | `sai_uint32_t` | 0 | Meter class of this meter bucket. |
+
+To fetch the metering data from each meter bucket, we are going to leverage the SAI stats APIs, which provides get, get and clear and other frequently used semantics. It will also reduce the work in SONiC stack, as SONiC already have good support over the stats APIs.
+
+| Attribute | Description |
+| --- | --- |
+| SAI_METER_BUCKET_STAT_OUTBOUND_BYTES | Total outbound traffic in bytes. |
+| SAI_METER_BUCKET_STAT_INBOUND_BYTES | Total inbound traffic in bytes. |
+
+### 3.2. Policy-based meter attributes
+
+#### 3.2.1. Route attributes
+
+| Attribute | Type | Default Value | Description |
+| --- | --- | --- | --- |
+| SAI_OUTBOUND_ROUTING_ENTRY_ATTR_METER_CLASS_OR | `sai_uint32_t` | 0 | Meter class OR bits. |
+| SAI_OUTBOUND_ROUTING_ENTRY_ATTR_METER_CLASS_AND | `sai_uint32_t` | UINT32_MAX | Meter class AND bits. |
+
+#### 3.2.2. Mapping attributes
+
+| Attribute | Type | Default Value | Description |
+| --- | --- | --- | --- |
+| SAI_OUTBOUND_CA_TO_PA_ENTRY_ATTR_METER_CLASS_OR | `sai_uint32_t` | 0 | Meter class OR bits. |
+
+#### 3.2.3. Tunnel attributes
+
+| Attribute | Type | Default Value | Description |
+| --- | --- | --- | --- |
+| SAI_DASH_TUNNEL_ATTR_METER_CLASS_OR | `sai_uint32_t` | 0 | Meter class OR bits. |
+
+#### 3.2.4. Inbound route rule attributes
+
+| Attribute | Type | Default Value | Description |
+| --- | --- | --- | --- |
+| SAI_INBOUND_ROUTING_ENTRY_ATTR_METER_CLASS_OR | `sai_uint32_t` | 0 | Meter class OR bits. |
+| SAI_INBOUND_ROUTING_ENTRY_ATTR_METER_CLASS_AND | `sai_uint32_t` | UINT32_MAX | Meter class AND bits. |
+
+### 3.3. Global meter policy
+
+#### 3.3.1. ENI
+
+| Attribute | Type | Default Value | Description |
+| --- | --- | --- | --- |
+| SAI_ENI_ATTR_V4_METER_POLICY_ID | `sai_object_id_t` | SAI_NULL_OBJECT_ID | Global IPv4 meter policy ID for this ENI. |
+| SAI_ENI_ATTR_V6_METER_POLICY_ID | `sai_object_id_t` | SAI_NULL_OBJECT_ID | Global IPv6 meter policy ID for this ENI. |
+
+#### 3.3.2. Meter policy
+
+| Attribute | Type | Default Value | Description |
+| --- | --- | --- | --- |
+| SAI_METER_POLICY_ATTR_IP_ADDR_FAMILY | `sai_ip_addr_family_t` | SAI_IP_ADDR_FAMILY_IPV4 | IP address family of the metering policy |
+
+#### 3.3.3. Meter rule
+
+| Attribute | Type | Default Value | Description |
+| --- | --- | --- | --- |
+| SAI_METER_RULE_ATTR_METER_POLICY_ID | `sai_object_id_t` | SAI_NULL_OBJECT_ID | Meter policy ID of this meter rule. |
+| SAI_METER_RULE_ATTR_METER_CLASS | `sai_uint32_t` | UINT32_MAX | Meter class when this meter rule is hit. |
+| SAI_METER_RULE_ATTR_DIP | `sai_ip_address_t` | NA | Destination IP for ternary match. |
+| SAI_METER_RULE_ATTR_DIP_MASK | `sai_ip_address_t` | NA | Destination IP mask for ternary match. |
```

Comment:
Any reason why it's ternary and not LPM?

Reply:
this is because priority is being used for crafting the metering rules.

Reply:
But are they prefixes or arbitrary bits?

Reply:
they are indeed prefixes, not arbitrary bits.